### ISSUE TITLE
Configurable default type mappings

### ISSUE THREAD

**Issue by jburzynski** [jburzynski/TypeGen#36]
Would be great if it was possible to inject the ITypeService instance or if there were some other way to affect the generated types. For example, for DateTime I'd like the TypeScript files to have the type string, and guid should map to string.

If I could provide ITypeService I wouldn't need to annotate all my source code. :)

I also need to provide TS generation for nuget packages, which I can't necessarily easily add type annotations to. Providing type mapping by convention would allow me to solve that locally.

_Originally posted by @maxmalmgren in https://github.com/jburzynski/TypeGen/issues/35#issuecomment-442071685_

**Comment by jburzynski** [comment]
It seeems like this (configurable default type mappings) can be a useful feature. If it's easy enough to implement, maybe it will be in the next release. I'll see what I can do and keep you updated.

BTW. Guid will be by default translated to string in the next version

**Comment by jburzynski** [comment]
@maxmalmgren I updated the NuGet packages in the [dev branch](https://github.com/jburzynski/TypeGen/tree/dev) (both regular and .NET CLI tool; `nuget` and `nuget-dotnetcli` folders). They now contain the functionality to map any C# type to a TypeScript type. You would use it like this in `tgconfig.json` (there's also a corresponding option in `GenerationOptions`):

```json
{
  "customTypeMappings": { "System.DateTime": "string", "System.DateTimeOffset": "string" }
}
```

Can you please check if it works properly in your case? I'd like to make sure it works before releasing. Thanks

Also, Guid is now mapped by default to string, so you don't need to specify it in `customTypeMappings`.

**Comment by jburzynski** [comment]
I've released version 2.0.1 (major version change is because of the changed syntax in the `TypeGen` nuget package - it's now the same as in the .NET CLI tool) and it now contains the functionality to define custom type mappings. The usage is exactly as in my previous post:

```json
{
  "customTypeMappings": { "System.DateTime": "string", "System.DateTimeOffset": "string" }
}
```

Also, Guid is now mapped by default to string, so it doesn't have to be in `customTypeMappings`.
